lib/Stage/CopyInputFastq.pm

```diff
@@ -23,9 +23,30 @@ sub new
         return $self;
 }
 
+#Method to verify that all input fastq files and the reference fasta file have unique file names.
+#If duplicate names are found, die is called and an error message is logged/printed to STDERR.
+#input:
+#	$fastq_names: Array of Strings describing the URL for each input fastq file.
+#	$ref_name: URL of the input reference.fasta file.
+sub verify_unique_file_names
+{
+	my ($self, $fastq_names, $ref_name) = @_;
+	my %hashCounter=();
+	$hashCounter{basename($ref_name, '.fasta')} = 1;
+    for my $name (@$fastq_names)
+	{
+		if(defined $hashCounter{ basename($name, '.fastq') })
+		{
+			print STDERR "Error: Duplicate file name found in fastq/reference input files. Please rename the file: $name.fastq\n";
```

Comment:
Note: `die` will also print to `STDERR` so you don't need to duplicate messages here.  However, could you also print this message using my `$logger` object similar to the code in the `execute` method?  This will log it to a file so it can be reviewed what went wrong later.